sra_experiment/nbconverted/DEG_validation.py

```diff
@@ -0,0 +1,311 @@
+
+# In[2]:
+
+
+# Download sra data files
+get_ipython().system(' prefetch --option-file $paths.SRA_ACC_TEST')
+
+
+# In[3]:
+
+
+get_ipython().run_cell_magic('bash', '', 'mkdir $paths.FASTQ_TEST_DIR\nfastq-dump $paths.SRA_DIR/SRR8486287.sra --outdir $paths.FASTQ_TEST_DIR/\nfastq-dump $paths.SRA_DIR/SRR8486288.sra --outdir $paths.FASTQ_TEST_DIR/\nfastq-dump $paths.SRA_DIR/SRR8486289.sra --outdir $paths.FASTQ_TEST_DIR/\nfastq-dump $paths.SRA_DIR/SRR8486290.sra --outdir $paths.FASTQ_TEST_DIR/')
+
+
+# In[4]:
+
+
+get_ipython().run_cell_magic('bash', '-s $paths.PAO1_QUANT_TEST $paths.FASTQ_TEST_DIR $paths.PAO1_INDEX', 'mkdir $1\n\nfor FILE_PATH in $2/*;\ndo\n\n# get file name\nsample_name=`basename ${FILE_PATH}`\n\n# remove extension from file name\nsample_name="${sample_name%_*}"\n\n# get base path\nbase_name=${FILE_PATH%/*}\n\necho "Processing sample ${sample_name}"\n\nsalmon quant -i $3 -l A \\\n            -r ${base_name}/${sample_name} \\\n            -p 8 --validateMappings -o $1/${sample_name}_quant\ndone')
+
+
+# In[5]:
+
+
+# Get raw read counts using PAO1 reference
+# Read through all sample subdirectories in quant/
+# Within each sample subdirectory, get quant.sf file
+data_dir = paths.PAO1_QUANT_TEST
+
+expression_data = pd.DataFrame(
+    pd.read_csv(file, sep="\t", index_col=0)["NumReads"].
+    rename(file.parent.name.split("_")[0]) 
+    for file in data_dir.rglob("*/quant.sf"))    
+
+# Map gene ids to gene names
+pao1_fasta_file = paths.PAO1_REF
+
+seq_id_to_gene_id_pao1 = utils.dict_gene_num_to_ids(pao1_fasta_file)
+
+expression_data.rename(mapper=seq_id_to_gene_id_pao1, axis="columns", inplace=True)
+
+expression_data.head()
+
+
+# In[6]:
+
+
+new_index = [name.split(".")[0] for name in list(expression_data.index)]
+expression_data.index = new_index
+expression_data.head()
+
+
+# ### Process data
+# 1. Get core genes
+# 2. Round read counts to integer value
+
+# In[7]:
+
+
+# Get mapping between PAO1 and PA14 genes using PAO1 reference
+gene_annot_file = paths.GENE_PAO1_ANNOT
+gene_mapping_pao1 = utils.get_pao1_pa14_gene_map(gene_annot_file, 'pao1')
+gene_annot_file = paths.GENE_PA14_ANNOT
+gene_mapping_pa14 = utils.get_pao1_pa14_gene_map(gene_annot_file, 'pa14')
+
+core_pao1_genes, core_pa14_genes = utils.get_core_genes(gene_mapping_pao1,
+                                                        gene_mapping_pa14,
+                                                        False)
+print(f"Number of PAO1 core genes: {len(core_pao1_genes)}")
+print(f"Number of PA14 core genes: {len(core_pa14_genes)}")
+
+core_pao1_genes = set(core_pao1_genes) - set(["PA4215", "PA4214","PA4213"])
+
+expression_data = expression_data.reindex(columns=core_pao1_genes)
+print(expression_data.shape)
+expression_data.head()
+
+
+# In[8]:
+
+
+# Convert values to integers for DE analysis
+# Not sure why the "Numreads" values are floats
```

Comment:
You're correct. Salmon produced estimated counts but DESeq requires integers so there is some inconsistency there, but looks like its a known problem. Looks like if I had used tximport (which maps transcript-level expression to gene-level expression), it would have returned counts, but since I was already at the gene-level I skipped this step and thus the estimated counts that I manually rounded. But this brings up a good point that I need to look more carefully to see what tximport and DESeq are doing in terms of processing the data to make sure my rounding isn't doing something it shouldn't